### PR TITLE
LinkedElementsInView, CurveLoop Tessellate, View to Solid methods

### LINKED ISSUE #1202: Get the control points of the Curve Loop
#### Description:

As in the title.

### PR DESCRIPTION
**Pull request by michal-pekacki**
### Issues addressed by this PR

Closes #1202 

- added `LinkedElementsInView` method to get linked elements from given view
- added `Tessalate` method to convert `CuveLoop` to list of points
- extend `Solid` method to get the `Solid` from `ViewPlan`

### Additional comments

New PR for #1230 
This PR has been closed (new PR - #1242)

### CODE REVIEW

**Comment by michal-pekacki** [comment]
@BHoMBot check compliance

**Comment by michal-pekacki** [comment]
@BHoMBot check compliance

File: Revit_Core_Engine/Create/Solid.cs [hunk 1]
```diff
@@ -94,5 +94,62 @@ public static Solid Solid(IList<CurveLoop> boundaries, double bottomElevation, d
...
+                double bottomOffset = viewRange.GetOffset(PlanViewPlane.BottomClipPlane);
+                bottomElevation = bottomLevel.ProjectElevation + bottomOffset;
+            }
+
+            viewBBox.Min = new XYZ(viewBBox.Min.X, viewBBox.Min.Y, bottomElevation);
+            viewBBox.Max = new XYZ(viewBBox.Max.X, viewBBox.Max.Y, topElevation);
+            viewBBox.Transform = Transform.Identity;
+
+            Solid viewSolid = viewBBox.ToSolid();
+            Solid transformedSolid = SolidUtils.CreateTransformed(viewSolid, transform.Inverse);
```
Comment (pawelbaran) [inline Revit_Core_Engine/Create/Solid.cs:148]: Sanity check: the inverse transform is what is needed here? It always gets a bit confusing 😃

File: Revit_Core_Engine/Query/LinkedElementsInView.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        /***************************************************/
+
+        [Description("Get elements from link instance located in given view. Method returns all elements in the view scope (including hidden elements).")]
+        [Input("view", "ViewPlan to get the elements from.")]
+        [Input("linkInst", "Revit Link Instance to get the elements from.")]
+        [Input("elementFilters", "ElementFilter list for the elements.")]
+        [Output("elements", "Elements of link instance in given view.")]
+        public static List<Element> LinkedElementsInView(this ViewPlan view, RevitLinkInstance linkInst, List<ElementFilter> elementFilters)
+        {
+            if (view == null || linkInst == null || elementFilters.Count == 0)
```
Comment (pawelbaran) [inline Revit_Core_Engine/Query/LinkedElementsInView.cs:44]: what if `elementFilters == null`?

File: Revit_Core_Engine/Query/LinkedElementsInView.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    public static partial class Query
+    {
+        /***************************************************/
+        /****              Public methods               ****/
+        /***************************************************/
+
+        [Description("Get elements from link instance located in given view. Method returns all elements in the view scope (including hidden elements).")]
+        [Input("view", "ViewPlan to get the elements from.")]
+        [Input("linkInst", "Revit Link Instance to get the elements from.")]
+        [Input("elementFilters", "ElementFilter list for the elements.")]
```
Comment (pawelbaran) [inline Revit_Core_Engine/Query/LinkedElementsInView.cs:40]: What do you mean? This should be better explained. Also what if I do not want any extra filtering being applied and leave the value as `null`?

File: Revit_Core_Engine/Create/Solid.cs [hunk 4]
```diff
@@ -94,5 +94,62 @@ public static Solid Solid(IList<CurveLoop> boundaries, double bottomElevation, d
...
 
         /***************************************************/
+
+        [Description("Create Solid object from ViewPlan (based on cropbox and view range).")]
+        [Input("view", "ViewPlan to get the solid from.")]
+        [Input("transform", "Transform of the link model.")]
+        [Output("solid", "Solid object created from the ViewPlan.")]
+        public static Solid Solid(this ViewPlan view, Transform transform)
+        {
+            if (view == null || transform == null)
```
Comment (pawelbaran) [inline Revit_Core_Engine/Create/Solid.cs:104]: I do not get it: can't i create the solid from view if it is not in link?

File: Revit_Core_Engine/Query/LinkedElementsInView.cs [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        [Output("elements", "Elements of link instance in given view.")]
+        public static List<Element> LinkedElementsInView(this ViewPlan view, RevitLinkInstance linkInst, List<ElementFilter> elementFilters)
```
Comment (pawelbaran) [inline Revit_Core_Engine/Query/LinkedElementsInView.cs:42]: Please try to use full names in the method parameters
```suggestion
        public static List<Element> LinkedElementsInView(this ViewPlan view, RevitLinkInstance linkInstance, List<ElementFilter> elementFilters)
```

**Review by pawelbaran — CHANGES_REQUESTED** [review]

File: Revit_Core_Engine/Create/Solid.cs (hunk 1 above)
Comment (michal-pekacki) [reply Revit_Core_Engine/Create/Solid.cs:148]: Me too, but I checked it again and yes it is needed 😃

File: Revit_Core_Engine/Query/LinkedElementsInView.cs (hunk 3 above)
Comment (michal-pekacki) [reply Revit_Core_Engine/Query/LinkedElementsInView.cs:40]: Good idea, I added return for `elementFilters = null`

File: Revit_Core_Engine/Create/Solid.cs (hunk 4 above)
Comment (michal-pekacki) [reply Revit_Core_Engine/Create/Solid.cs:104]: Changed it to work in this case.

**Comment by michal-pekacki** [comment]
@BHoMBot check compliance

**Comment by michal-pekacki** [comment]
@BHoMBot check required

**Comment by michal-pekacki** [comment]
@BHoMBot check null-handling

**Comment by michal-pekacki** [comment]
@BHoMBot check null-handling

**Review by pawelbaran — APPROVED** [review]
Looks good to me after all the tweaks that have been made. Happy to approve 👍

**Comment by pawelbaran** [comment]
@BHoMBot check ready-to-merge

**Comment by pawelbaran** [comment]
@BHoMBot check ready-to-merge

**Comment by michal-pekacki** [comment]
@BHoMBot check required

**Pull request closed without merging**